### ISSUE TITLE
Add baggage

### ISSUE THREAD

**Issue by pavolloffay** [openzipkin/b3-propagation#22]
Hello,

did you consider adding baggage to B3? Brave and I believe Sleuth (even before using Brave internally) support baggage with the following format:

`baggage-<key>: <value>`

**Comment by codefromthecrypt** [comment]
we have considered this but..

baggage
<https://groups.google.com/d/msg/distributed-tracing/m2ztbNAPsjY/4-mQkrfvBwAJ>
is
wider than tracing, even more narrowly defined it is. B3 is a tracing
standard, similar to trace-context
<https://www.w3.org/community/trace-context/>
More relevant to what you mention is correlation-context
<https://github.com/w3c/distributed-tracing/tree/master/correlation_context>
and census tags
<https://github.com/census-instrumentation/opencensus-specs/blob/master/tags/TagContext.md>
both
of which are strict on size etc, decoupled from tracing as they are used in
metrics etc.
random prefixed keys have many problems, which is why amazon and w3c work
don't do that. You can look at correlation-context for more.
The only example of mixing the two (trace and correlation fields)
effectively is Amazon's trace format
<https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-request-tracing.html>,
still in a single header though.

TL;DR; is that B3 isn't going to implement known malpractice (prefixed
fields), and rather rely on other formats for things like this.
It doesn't make sense for us to make a one-off version of what's being done
in correlation-context.

**Comment by codefromthecrypt** [comment]
while completely unrelated to this repo, those interested in the sleuth/brave-specific question..

Brave allows through a tool called ExtraFieldsPropagation to push predefined fields across the wire. It doesn't matter if they are prefixed or not, but random unknown fields are not supported. Census has strict rules on size (jaeger also has size management tools) and there are discussions about privacy and leak concerns. Basically the "baggage-X" stuff was rushed and not a good idea to use the way it is today IMHO. I think paying attention to the correlationcontext spec (and also John Mace's research) is higher value than the "baggage" legacy stuff in OT.

Correlation fields (in Brave) will be separate functionality and bound to other specs like amazon and correlation-context via [this issue](https://github.com/openzipkin/brave/pull/577). Tracing doesn't need to own "baggage" and in fact sometimes that's a problem as tags can be defined before tracing starts. This is why census keeps them separate.

**Comment by pavolloffay** [comment]
@adriancole thanks for pointers and explanation. The correlation-context nicely explains its benefits compared to prefixed keys https://github.com/w3c/distributed-tracing/blob/master/correlation_context/HTTP_HEADER_FORMAT_RATIONALE.md#why-a-single-header. 

I see that brave does not mandate to use `baggage-` prefix. It has to be configured explicitly. Which is good for the transition to correlation-context. However at the moment probably majority users use `baggage-` prefix anyways.

Sleuth, on the other hand, uses `baggage-` implicitly. Which means majory of the users using B3 probably use `baggage-` prefix (If they use baggage).

Given above I would expect its mentioned somewhere, maybe as an optional thing - like it is in w3c? 

Do you expect people using B3 with correlation-context? Correlation-context will be out after trace-context so probably people will start using correlation-context with trace-context and not B3.

**Comment by bruce-szalwinski** [comment]
https://github.com/w3c/trace-context/blob/main/http_header_format_rationale.md update link.  `master` -> `main`
